### ISSUE TITLE
Deprecation warnings don't specify the iOS version

### ISSUE THREAD

**Issue by Reflejo** [Appboy/appboy-ios-sdk#89]
When building for iOS 8+ we are getting a warning:

```
`registerApplication:didReceiveRemoteNotification:` is deprecated in iOS 10, please use `registerApplication:didReceiveRemoteNotification:fetchCompletionHandler:` instead.
```

which seems to be only the case for iOS 10+; seems like the `@available` call is missing the iOS version.

Also:

```
'getActionWithIdentifier(_:forRemoteNotification:completionHandler:)' is deprecated: `getActionWithIdentifier:forRemoteNotification:completionHandler:` is deprecated in iOS 10, please use `userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:` instead.
```

**Comment by Wenzhi** [comment]
Hey @Reflejo,

Thanks for reporting the issue and suggesting the `@available` solution! Unfortunately, `@available` isn’t available in Objective-C, in which our SDK is written. There is no equivalent option in objective-C. We are using `__deprecated_msg()` to mark methods deprecated so it just generates a warning, but the methods are still available and people can use them.

You can ignore the warning for now. Please let us know if you have any other questions.

Thanks,
Wenzhi

**Comment by Reflejo** [comment]
@Wenzhi Sadly ignoring the warnings is not an option for us (tbh it shouldn't be an option for anyone!) because we threat warnings as errors and we'll have to call this method for at least 1 more year since we need to support < iOS 10. You guys can use `NS_DEPRECATED_IOS` on objective-c such as:

```objc
- (void)onlyForiOS9 NS_DEPRECATED_IOS(6_0, 10_0, "Use iOS10 instead");
```

That way this would only alert projects targeting iOS10+.

![screen shot 2016-10-14 at 10 31 10 am](https://cloud.githubusercontent.com/assets/232113/19396765/897f63b6-91f9-11e6-8961-4bce45ffc3bc.png)

**Comment by Wenzhi** [comment]
@Reflejo 

Thanks for the suggestion! We will apply it in our next release.

Cheers,
Wenzhi

**Comment by Wenzhi** [comment]
Hey @Reflejo, 

We just released SDK 2.24.3 with the fix; please check it out! I'm going to close this issue, but please feel free to re-open it if you still see the issue in the new release.

Thanks,
Wenzhi
